src/Mod/Sketcher/Gui/CommandConstraints.cpp

```diff
@@ -2098,6 +2098,23 @@ class DrawSketchHandlerDimension : public DrawSketchHandler
                 Gui::cmdAppObjectArgs(Obj, "addConstraint(Sketcher.Constraint('Distance',%d,%d,%f))",
                     GeoId1, GeoId2, ActDist);
             }
+            if (isArcOfCircle(*geo1) && isLineSegment(*geo2)) { // ArcOfCircle - line case
```

Comment:
In this case there is a lot of copy paste with previous mode. You can improve like so : 
```
            if ((isCircle(*geo1) || isArcOfCircle(*geo1)) && isLineSegment(*geo2)) { // Circle/arc - line case
                double radius;
                Base::Vector3d center;
                if (isCircle(*geo1)) {
                    auto conic = static_cast<const Part::GeomCircle*>(geo1);
                    radius = conic->getRadius();
                    center = conic->getCenter();
                }
                else {
                    auto conic = static_cast<const Part::GeomArcOfCircle*>(geo1);
                    radius = conic->getRadius();
                    center = conic->getCenter();
                }

                auto lineSeg = static_cast<const Part::GeomLineSegment*>(geo2);
                Base::Vector3d pnt1 = lineSeg->getStartPoint();
                Base::Vector3d pnt2 = lineSeg->getEndPoint();
                Base::Vector3d d = pnt2 - pnt1;
                double ActDist =
                    std::abs(-center.x * d.y + center.y * d.x + pnt1.x * pnt2.y - pnt2.x * pnt1.y)
                    / d.Length()
                    - radius;

                Gui::cmdAppObjectArgs(Obj, "addConstraint(Sketcher.Constraint('Distance',%d,%d,%f))",
                    GeoId1, GeoId2, ActDist);
            }
```